.github/workflows/commit-access-review.py

```diff
@@ -0,0 +1,443 @@
+            for pr in data["search"]["nodes"]:
+                # Users can be None if the user has been deleted.
+                if not pr["author"]:
+                    continue
+                author = pr["author"]["login"]
+                if author in triage_list:
+                    triage_list[author].add_authored()
+
+                if not pr["mergedBy"]:
+                    continue
+                merger = pr["mergedBy"]["login"]
+                if author == merger:
+                    continue
+                if merger not in triage_list:
+                    continue
+                triage_list[merger].add_merged()
+
+            has_next_page = data["search"]["pageInfo"]["hasNextPage"]
+            if has_next_page:
+                variables["after"] = data["search"]["pageInfo"]["endCursor"]
+        date_begin = date_end
+
+
+def main():
+    token = sys.argv[1]
+    gh = github.Github(login_or_token=token)
+    org = gh.get_organization("llvm")
+    repo = org.get_repo("llvm-project")
+    team = org.get_team_by_slug("llvm-committers")
+    one_year_ago = datetime.datetime.now() - datetime.timedelta(days=365)
+    triage_list = {}
+    for member in team.get_members():
+        triage_list[member.login] = User(member.login, triage_list)
+
+    print("Start:", len(triage_list), "triagers")
+    # Step 0 Check if users have requested commit access in the last year.
+
+    for user in check_manual_requests(one_year_ago, token):
+        if user in triage_list:
+            print(user, "requested commit access in the last year.")
+            del triage_list[user]
+    print("After Request Check:", len(triage_list), "triagers")
+
+    # Step 1 count all PRs authored or merged
+    count_prs(triage_list, one_year_ago, token)
+
+    print("After PRs:", len(triage_list), "triagers")
+
+    if len(triage_list) == 0:
+        sys.exit(0)
+
+    # Step 2 check for reviews
+    for user in list(triage_list.keys()):
+        review_count = get_review_count(user, one_year_ago, token)
+        triage_list[user].add_reviewed(review_count)
+
+    print("After Reviews:", len(triage_list), "triagers")
+
+    if len(triage_list) == 0:
+        sys.exit(0)
+
+    # Step 3 check for number of commits
+    for user in list(triage_list.keys()):
+        num_commits = get_num_commits(user, one_year_ago, token)
+        # Override the total number of commits to not double count commits and
+        # authored PRs.
+        triage_list[user].set_authored(num_commits)
+
+    print("After Commits:", len(triage_list), "triagers")
+
+    # Step 4 check for new committers
+    for user in list(triage_list.keys()):
+        print("Checking", user)
+        if is_new_committer(user, one_year_ago, token):
+            print("Removing new committer: ", user)
+            del triage_list[user]
+
+    print("Complete:", len(triage_list), "triagers")
+
+    filename = "triagers.log"
+    f = open(filename, "w")
```

Comment:
Put this into a `with` statement to avoid manually having to call the `f.close()`? Also, a more descriptive name would probably be useful.